ivy/functional/frontends/tensorflow/math.py

```diff
@@ -612,6 +612,28 @@ def scalar_mul(scalar, x, name="scalar_mul"):
     return ivy.multiply(x, scalar).astype(x.dtype)
 
 
+@to_ivy_arrays_and_back
+def segment_max(data, segment_ids, name="segment_max"):
+    data = ivy.array(data)
```

Comment:
in the frontend, you should just return `ivy.segment_max()`.
and you should add this block of code to `ivy.functional.ivy.activations.py` instead. please have a look at [https://unify.ai/docs/ivy/overview/deep_dive/function_types.html#mixed-functions](https://unify.ai/docs/ivy/overview/deep_dive/function_types.html#mixed-functions)

Reply:
I have made changes in `ivy.functional.ivy.activation.py` as you instructed